Stable nitrogen

santanude opened the pull request:
Create pull request for code review and merge Legato module to Donald's repository

santanude committed:
commit Legato changes

santanude committed:
commit Legato changes

santanude committed:
commit Legato changes

santanude committed:
commit Legato changes

santanude committed:
Merge branch 'stable-nitrogen' of https://github.com/donaldh/unimgr into stable-nitrogen

ahussain786 committed:
L2cp service added

ahussain786 committed:
L2cp service added

ahussain786 committed:
L2cp EEC and Peering service add

ahussain786 committed:
L2cp EEC and Peering service add

ahussain786 committed:
Modification: delete EVC node from OPERATIONAL DB method calling from
LegatoUtils

ahussain786 committed:
Modification: delete EVC node from OPERATIONAL DB method calling from
LegatoUtils

Arnav2003 committed:
BWP ColorMapping Cos EEC profile controllers

Arnav2003 committed:
BWP ColorMapping Cos EEC profile controllers

Arnav2003 committed:
BWP ColorMapping Cos EEC profile controllers

Arnav2003 committed:
BWP ColorMapping Cos EEC profile controllers

Arnav2003 committed:
Merge branch 'stable-nitrogen' of https://github.com/santanude/unimgr into stable-nitrogen

ahussain786 committed:
Modification: Used generic add and delete method

ahussain786 committed:
Modification: Used generic add and delete method

Arnav2003 committed:
Final code after suggested changes

Arnav2003 committed:
Final code after suggested changes

santanude committed:
commit EVPL test case

santanude committed:
commit EVPL test case

mjethanandani commented on legato-api/pom.xml:
```diff
@@ -7,17 +7,28 @@ terms of the Eclipse Public License v1.0 which accompanies this distribution,
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-  <parent>
+
+<!--  <parent>
```
In general we should avoid crud in our code. If this part of the configuration is not required, it is better to remove it. Git will take care of recovering it if needed.

mjethanandani commented on legato-api/pom.xml:
```diff
@@ -41,11 +52,31 @@ and is available at http://www.eclipse.org/legal/epl-v10.html INTERNAL
     </dependency>
-    <dependency>
+    
+    <!--<dependency>
```
Same comment as above for crud.

mjethanandani commented on legato-api/src/main/java/org/opendaylight/legato/LegatoCosProfileController.java:
```diff
@@ -0,0 +1,243 @@
+	private static final InstanceIdentifier<Profile> PROFILE_IID = InstanceIdentifier
+			.builder(MefGlobal.class).child(CosProfiles.class).child(Profile.class).build();
+	
+	
```
Have you seen these warnings during compilation?

[WARN] /misc/data/users/mahesh/src/git/unimgr/legato-api/src/main/java/org/opendaylight/legato/LegatoCosProfileController.java:52: 'method def modifier' have incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /misc/data/users/mahesh/src/git/unimgr/legato-api/src/main/java/org/opendaylight/legato/LegatoCosProfileController.java:52:1: Line contains a tab character. [FileTabCharacter]
[WARN] /misc/data/users/mahesh/src/git/unimgr/legato-api/src/main/java/org/opendaylight/legato/LegatoCosProfileController.java:53:1: Line contains a tab character. [FileTabCharacter]

mjethanandani commented on legato-api/src/main/java/org/opendaylight/legato/LegatoCosProfileController.java:
```diff
@@ -0,0 +1,243 @@
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+//import org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.color.mapping.profiles.ProfileKey;;
```
Unless you need this line, please remove all crud.

mjethanandani commented on features/odl-unimgr-legato-api/pom.xml:
```diff
@@ -0,0 +1,78 @@
+      
+    </dependency>
+      
+    <!--<dependency>
```
Remove if not needed.

mjethanandani commented on features/odl-unimgr-legato-api/pom.xml:
```diff
@@ -0,0 +1,78 @@
+      <version>${project.version}</version>
+    </dependency>-->  
+    
+    <!--
```
Same here.

mjethanandani commented on legato-api/src/main/java/org/opendaylight/legato/LegatoBwpProfileController.java:
```diff
@@ -0,0 +1,190 @@
+/*
```
Indentation in this file is off. You are using tabs instead of spaces for the file. That will inconsistent indentation depending on how tabs is defined. You have to use spaces, and that too 4, instead of 8.

mjethanandani commented on legato-api/src/main/java/org/opendaylight/legato/LegatoBwpProfileController.java:
```diff
@@ -0,0 +1,190 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
```
Check for a compiler warning message here about the order of import.

mjethanandani commented on legato-api/src/main/java/org/opendaylight/legato/LegatoBwpProfileController.java:
```diff
@@ -0,0 +1,190 @@
+	public LegatoBwpProfileController(DataBroker dataBroker) {
+		super(dataBroker);
+		registerListener();
+		// TODO Auto-generated constructor stub
```
Do you need this comment still?

mjethanandani commented on legato-api/src/main/java/org/opendaylight/legato/LegatoBwpProfileController.java:
```diff
@@ -0,0 +1,190 @@
+	private ListenerRegistration<LegatoBwpProfileController> dataTreeChangeListenerRegistration;
+
+	private void registerListener() {
+		LOG.info("Initializing LegatoSlsProfileController:init() ");
```
Is this initializing LegatoSlsProfileController or LegatoBwpProfileController?

bartoszm commented on legato-api/src/main/resources/org/opendaylight/blueprint/legato-api-blueprint.xml:
```diff
@@ -0,0 +1,46 @@
+   <odl:rpc-service id="prestoConnectivityService" interface="org.opendaylight.yang.gen.v1.urn.onf.params.xml.ns.yang.tapi.connectivity.rev171113.TapiConnectivityService" />
+ 
+   <bean id="mef-services" class="org.opendaylight.legato.LegatoServiceController" destroy-method="close">
+   		<argument ref="dataBroker" />
```
please make sure you are using one style of intends in the file 
e.g. for arguments properties  here you are using /t whereas for other places you use spaces.
please make sure it is consistent throughout the code (you can setup your IDE to do that either on the fly or at the commit)

bartoszm commented on legato-api/src/main/java/org/opendaylight/legato/util/LegatoConstants.java:
```diff
@@ -0,0 +1,29 @@
+	public static final String EVC_UNI_LIST = "EVC_UNI_LIST";
+	public static final String EVC_CON_TYPE = "EVC_CON_TYPE";
+	public static final String BEST_EFFORT = "BEST_EFFORT";
+	public static final long LONG_VAL = 9223372;
```
what is that?

bartoszm commented on legato-api/src/main/java/org/opendaylight/legato/LegatoSlsProfileController.java:
```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2014 Cisco Systems, Inc. and others.  All rights reserved.
```
2018 instead of 2014

bartoszm commented on legato-api/src/main/java/org/opendaylight/legato/LegatoSlsProfileController.java:
```diff
@@ -0,0 +1,165 @@
+	
+	@Override
+	public void add(DataTreeModification<Profile> newDataObject) {
+		if(newDataObject.getRootNode() != null && newDataObject.getRootPath() != null ){
```
as getRootNode and getRootPath are marked @Notnull you do not need to check that. if you want to stay on the safe side you can use *assert* 
you have it elsewhere in the file. please fix it in the whole code base for legato module

bartoszm commented on legato-api/src/main/java/org/opendaylight/legato/util/LegatoUtils.java:
```diff
@@ -0,0 +1,386 @@
+		WriteTransaction tx = dataBroker.newWriteOnlyTransaction();
+		tx.merge(LogicalDatastoreType.OPERATIONAL, profilesTx, typeOfProfile);
+		try {
+			tx.submit().checkedGet();
```
as you are not interested in result you do not need to check for submit result. you can use ``Futures.addCallback(``

bartoszm commented on legato-api/src/main/java/org/opendaylight/legato/LegatoServiceController.java:
```diff
@@ -0,0 +1,384 @@
+
+	private ListenerRegistration<LegatoServiceController> dataTreeChangeListenerRegistration;
+
+	private static final Map<String, String> EVC_UUIDMap = new HashMap<String, String>();
```
please use Java variable naming convention

bartoszm commented on legato-api/src/main/java/org/opendaylight/legato/LegatoServiceController.java:
```diff
@@ -0,0 +1,384 @@
+				&& modifiedDataObject.getRootPath() != null) {
+			LOG.info("  Node modified  "
+					+ modifiedDataObject.getRootNode().getIdentifier());
+			Optional<Evc> OptionalEvc = LegatoUtils.readEvc(dataBroker,
```
please use Java variable naming convention
should be ``optionalEvc`` please fix throughout the code

bartoszm commented on legato-api/src/main/java/org/opendaylight/legato/LegatoServiceController.java:
```diff
@@ -0,0 +1,384 @@
+		}
+	}
+	
+	private void UpdateEvcOnOperationalDB(Evc evc) {
```
please use Java naming convention for method names, should be ``updateEvcOnOperationalDb``
Please make sure you fix it in the whole pull request

bartoszm commented on legato-api/src/main/java/org/opendaylight/legato/LegatoServiceController.java:
```diff
@@ -0,0 +1,384 @@
+			LOG.info("  Node Added  "
+					+ newDataObject.getRootNode().getIdentifier());
+
+			Optional<Evc> OptionalEvc = LegatoUtils.readEvc(dataBroker,
```
Actually you do not need to do an extra lookup to MD-SAL -  tree modification has all data you need.
``Evc evc = newDataObject.getRootNode().getDataAfter();`` 
Please consider using this apporach elsewhere. The contract is:
* ``getDataAfter`` has value for create and update
* ``getDataBefore`` has value for delete and update
in case of update getDataBefore stores old and getDataAfter a new version of the object

bartoszm commented on legato-api/src/main/java/org/opendaylight/legato/util/LegatoUtils.java:
```diff
@@ -0,0 +1,386 @@
+				.setMaxFrameSize(new PositiveInteger(Long.parseLong(maxFrameSize))).build();
+	}
+
+	public static final CreateConnectivityServiceInput buildCreateConnectivityServiceInput(Map<String, Object> evcMap,
```
using ``Map<String,Object>`` for evc representation might be hard to maintain in the long run
* you need to know type of value under each key
* you need to use ton of casts
* you can learn that there is a problem only at runtime
* nobody will stop another person contributing to this code to make it in a wrong way

You can consider ``class EvcParamsDto`` when instead of key values you have typed attributes:

e.g. 
instead ``(List<String) EVC_UNI_LIST`` you have ``List<String> EvcParamsDto.getEvcUniList()``
instead ``evcMap.get(LC.EVC_MAX_FRAME).toString() you have ``long EvcParamsDto.getMaxFrameSize()``

bartoszm reviewed (changes requested):
I would suggest to change base package from 
org.opendaylight.legato to 
org.opendaylight.unimgr.mef.legato

Please make sure that copyright is correct for all files - it should state 2018 for new files.
Please make sure to add author to every single new file.

We need to have tests (unit and integration) added to legato module.

Please remove extra white-spaces from all files

bartoszm committed:
updating revision date for Presto NRP 21.03.2018 revision

Change-Id: I1df74058325577f4e5c3a451107266e4ca6e3047
Signed-off-by: Bartosz Michalik <Bartosz.Michalik@amartus.com>

bartoszm committed:
Refactoring - moving netconf related dependencies from core implementation to cisco-xr-driver

Change-Id: Ibd3f12624703c6947dcefa015fba366412b7762b
Signed-off-by: Bartosz Michalik <Bartosz.Michalik@amartus.com>

santanude committed:
fix line indentation

santanude committed:
fix line indentation

santanude committed:
pom.xml cleanup

santanude committed:
pom.xml cleanup

grzegorzkisiel committed:
VLAN translation + 'pool' for internal VLANs

grzegorzkisiel committed:
validation of external VLANs

santanude committed:
multipoint test documentation

santanude committed:
multipoint test documentation

santanude committed:
multipoint test documentation

santanude committed:
multipoint test documentation

santanude committed:
Delete EVPL_Test_Case_Doc.docx

santanude committed:
Delete EVPL_Test_Case_Doc.docx

grzegorzkisiel committed:
review remarks implemented

santanude committed:
Update legato-api-blueprint.xml

Update package path

santanude committed:
Update legato-api-blueprint.xml

Update package path

ahussain786 committed:
LEGATO API :: Integration and unit test cases committed.

ahussain786 committed:
LEGATO API :: Integration and unit test cases committed.

ahussain786 committed:
Update Evc Integration

ahussain786 committed:
Update Evc Integration

ahussain786 committed:
Package name changed

ahussain786 committed:
Package name changed

santanude committed:
refactore package

santanude committed:
refactore package

santanude committed:
copyright changes

santanude committed:
copyright changes

santanude committed:
copyright changes

santanude committed:
copyright changes

bartoszm committed:
Merge pull request #7 from grzegorzkisiel/stable-nitrogen

vlan translation

santanude committed:
fix line indentation

santanude committed:
fix line indentation

santanude commented:
Create new pull request

mjethanandani commented on legato-api/src/main/java/org/opendaylight/unimgr/mef/legato/LegatoColorMappingProfileController.java:
```diff
@@ -0,0 +1,120 @@
+    private void addToOperationalDB(Profile profile) {
+        try {
+            assert profile != null;
+        ColorMappingProfiles colorMappingProfiles = new ColorMappingProfilesBuilder().setProfile(Collections.singletonList(profile)).build();   
```
I still see indentation issues in the file, like on this line. Can we run indent on the entire file? If you use Eclipse, you can select the entire file and run indent on it.

mjethanandani commented on legato-api/src/main/java/org/opendaylight/unimgr/mef/legato/util/LegatoUtils.java:
```diff
@@ -0,0 +1,397 @@
+                .setCbs(new NaturalNumber(LegatoConstants.LONG_VAL)).setEir(new NaturalNumber(LegatoConstants.LONG_VAL))
+                .setEirMax(new NaturalNumber(LegatoConstants.LONG_VAL))
+                .setEbs(new NaturalNumber(LegatoConstants.LONG_VAL)).setCouplingFlag(true)
+                .setColorMode(ColorMode.COLORAWARE).setTokenRequestOffset(new NaturalNumber(LegatoConstants.LONG_VAL));
```
Should this be COLOR_AWARE instead of COLORAWARE?

mjethanandani commented on legato-api/src/main/java/org/opendaylight/unimgr/mef/legato/util/LegatoUtils.java:
```diff
@@ -0,0 +1,397 @@
+        List<LayerProtocol> layerList = new ArrayList<LayerProtocol>();
+
+        switch (connType.replace("-", "").toUpperCase()) {
+        case LegatoConstants.POINTTOPOINT:
```
Can we be consistent in the naming convention. Either use _ between multiple words in the constant, or not.

santanude committed:
refactor parseNodes()"
;

santanude committed:
refactor parseNodes()"
;

santanude committed:
refactor parseNodes()

santanude committed:
refactor parseNodes()

santanude committed:
refactor code

santanude committed:
refactor code

bartoszm committed:
Code quality improvement
Fixing some of the RPC bugs after migrating to new Presto NRP
Updating postman collections

Change-Id: Id78a5d0dfcb97f1202f002665bb6845dc55c3cbd
Signed-off-by: Bartosz Michalik <Bartosz.Michalik@amartus.com>

bartoszm committed:
Fixing Legato mapping after migrating to latest Presto NRP models

bartoszm committed:
Merge branch 'stable-nitrogen' of github.com:santanude/unimgr into stable-nitrogen

bartoszm commented:
Santanu could you please close this pull request and open a new one? As we have synced the changes to your repository it would be a way easier to see scope of the change with a new pull request. Please remember to address the suggestions from this pull request in the new one.

santanude committed:
refactore code

ahussain786 committed:
Update packages import & legatoUtils testcase

ahussain786 committed:
Method changed
